### ISSUE TITLE
Need help to convert caffemodel to matconvnet ！

### ISSUE THREAD

**Issue by zwx8981** [vlfeat/matconvnet#508]
Hi,  I want to convert a vgg-16 caffe model trained on Places dataset to matconvnet, however, as a windows user, I am not familiar with the bash command and just don't know how to use the import-caffe.py like the import-ref-models.sh. Anyone can help me ? Thank you in advance.

**Comment by lenck** [comment]
Hi, sorry, the conversion script was being tested only on Linux so we are not able to help with the Windows version. You may try to get Cygwin to be able to run the bash script, but there may occur some Python issues further on the line. Don't you have access to a Linux machine?

**Comment by zwx8981** [comment]
@lenck Hi, yes, I've installed an ubuntu 14.04 system on virtual machine, I just don't know how to use the conversion script.

**Comment by lenck** [comment]
Yeah, this is not officially supported part of MatConvNet as it a bit of a mess as Caffe is changing their protobufs all the time. Basically you need to know a bit of bash and look how the `vgg-16` is converted in the [conversion script](https://github.com/vlfeat/matconvnet/blob/master/utils/import-ref-models.sh). It really shouldn't be that hard to hack it, what may be tricky is to pick the correct the right `caffe-variant`...
Once you make the script work for our vgg-16 model it should be relatively easy to use for yours...

**Comment by vedaldi** [comment]
Just to provide some more context, adapting to a newer version of Caffe requires:

- compiling the new protobuf definition (this is mainly what defines a "caffe variant"; the protobuf definition is a file in the Caffe distribution)
- changing the script to source that variant of protnbuf instead (this is easy)
- changing/adapting any layer that was changed in Caffe, or adding new one as needed

**Comment by omrysendik** [comment]
Hi all,

Another (really basic) question:
Running import-caffe.py, I get the following message: ImportError: No module named layers.

I tried searching for a python module/package named Layers and also one called Caffe but couldn't.
Can anyone who has done so on Windows 7 please support?

**Comment by zeevikal** [comment]
@omrysendik try to change 'layers' with 'layer'. it should work.
